cmd/tsidp/tsidp.go

```diff
@@ -220,7 +350,24 @@ func (s *idpServer) authorize(w http.ResponseWriter, r *http.Request) {
 		clientID:    uq.Get("client_id"),
 	}
 
-	if r.URL.Path == "/authorize/localhost" {
+	// This URL is visited by the user who is being authenticated. If they are
+	// visiting the URL over Funnel, that means they are not part of the
+	// tailnet that they are trying to be authenticated for.
+	if isFunnelRequest(r) {
```

Comment:
I am not sure this is correct. If I am connected to the tailnet serving tsidp over funnel I will get unauthorized here. Do we need to do anything different rather than checking who for funnel requests as well?

Reply:
I think I am incorrect here and not able to reproduce.